spring-session/src/main/java/org/springframework/session/web/http/CrawlerManagerSessionStrategyWrapper.java

```diff
@@ -0,0 +1,154 @@
+package org.springframework.session.web.http;
+
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.session.Session;
+import org.springframework.session.SessionRepository;
+
+public class CrawlerManagerSessionStrategyWrapper implements MultiHttpSessionStrategy {
+	private static final Log log = LogFactory.getLog(CrawlerManagerSessionStrategyWrapper.class);
+	
+	private HttpSessionStrategy delegate;
+	
+	private final Map<String,String> clientIpSessionId =
+            new ConcurrentHashMap<String,String>();
```

Comment:
`ConcurrentMap` is used because the map cache could be accessed  from multiple threads concurrently. ` #putIfAbsent()` is not used as overwrites are allowed. overwrites are legal if for some reason cache is not purged of old session before creating new session.